data/src/history/manager.rs

```diff
@@ -507,6 +547,23 @@ impl Data {
     }
 }
 
+fn smart_filter_message(
+    message: &crate::Message,
+    seconds: &i64,
+    most_recent_message_server_time: Option<&DateTime<Utc>>,
+) -> bool {
+    if most_recent_message_server_time.is_some() {
+        let duration_seconds = message
+            .server_time
+            .signed_duration_since(*most_recent_message_server_time.unwrap())
+            .num_seconds();
+
+        duration_seconds > *seconds
+    } else {
+        true
+    }
```

Comment:
Perhaps you could use let else here:
(psuedo code)

```rust
    let Some(server_time) = most_recent_message_server_time else {
        return true
    };

    let duration_seconds = message.server_time.signed_duration_since(*server_time).num_seconds();
    duration_seconds > *seconds
```

Reply:
Looks cleared to me, thanks for the suggestion.  Incorporated :)